tools/PowerShell/Microsoft.WinGet.Client/src/Library/Install-WinGetPackage.ps1

```diff
@@ -0,0 +1,174 @@
+        Used to specify the Moniker of the package
+
+        .PARAMETER Tag
+        Used to specify the Tag of the package
+        
+        .PARAMETER Command
+        Used to specify the Command of the package
+
+        .PARAMETER Scope
+        Used to specify install scope (user or machine)
+        
+        .PARAMETER Exact
+        Used to specify an exact match for any parameters provided. Many of the other parameters may be used for case insensitive substring matches if Exact is not specified.
+
+        .PARAMETER Source
+        Name of the Windows Package Manager private source. Can be identified by running: "Get-WinGetSource" and using the source Name
+
+        .PARAMETER Interactive
+        Used to specify the installer should be run in interactive mode.
+
+        .PARAMETER Silent
+        Used to specify the installer should be run in silent mode with no user input.
+
+        .PARAMETER Locale
+        Used to specify the locale for localized package installer.
+
+        .PARAMETER Log
+        Used to specify the location for the log location if it is supported by the package installer.
+
+        .PARAMETER Header
+        Used to specify the value to pass as the "Windows-Package-Manager" HTTP header for a REST source.
+
+        .PARAMETER Version
+        Used to specify the Version of the package
+
+        .PARAMETER Channel
+        Used to specify the Channel for the package. Note the Windows Package Manager as of version 1.1.0 does not support channels.
+
+        .PARAMETER VerboseLog
+        Used to provide verbose logging for the Windows Package Manager.
+        
+        .PARAMETER AcceptPackageAgreement
+        Used to accept any package agreement required for the package.
+        
+        .PARAMETER AcceptSourceAgreement
+        Used to explicitly accept any agreement required by the source.
+
+        .EXAMPLE
+        Install-WinGetPackage -id "Publisher.Package"
+
+        This example expects only a single package containing "Publisher.Package" as a valid identifier.
+
+        .EXAMPLE
+        Install-WinGetPackage -id "Publisher.Package" -source "Private"
+
+        This example expects the source named "Private" contains a package with "Publisher.Package" as a valid identifier.
+
+        .EXAMPLE
+        Install-WinGetPackage -Name "Package"
+
+        This example expects the source named "Private" contains a package with "Package" as a valid name.
+    #>
+
+    PARAM(
+        [Parameter(Position=0)] $Filter,
+        [Parameter()]           $Name,
+        [Parameter()]           $Id,
+        [Parameter()]           $Moniker,
+        [Parameter()]           $Source,
+        [Parameter()] [ValidateSet("User", "Machine")] $Scope,
+        [Parameter()] [switch]  $Interactive,
+        [Parameter()] [switch]  $Silent,
+        [Parameter()] [string]  $Version,
+        [Parameter()] [switch]  $Exact,
+        [Parameter()] [switch]  $Override,
+        [Parameter()] [string]  $Location,
```

Comment:
```suggestion
        [Parameter()] [System.IO.FileInfo]  $Location,
```
Using `[System.IO.FileInfo]` will automatically validate that the path could be valid even if it doesn't exist. I'll not mark everywhere this is applicable
```
PS C:\WINDOWS\system32> testthis "D:\?"
testthis : Cannot process argument transformation on parameter 'Path'. Cannot convert value "D:\?" to type
"System.IO.FileInfo". Error: "Illegal characters in path."
At line:1 char:10
+ testthis "D:\?"
+          ~~~~~~
    + CategoryInfo          : InvalidData: (:) [testthis], ParameterBindingArgumentTransformationException
    + FullyQualifiedErrorId : ParameterArgumentTransformationError,testthis

PS C:\WINDOWS\system32> testthis "D:\"

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d--hs-        11/14/2021   7:37 PM

PS C:\WINDOWS\system32> testthis "D:\ThisDoesntExist"

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
darhsl        12/31/1600   6:00 PM             () ThisDoesntExist
```

Reply:
Theres a few other places this could be a good change to add @denelon 

Theres a few locations that $Log is referenced that I didn’t mark with suggestions